src/app/shared/custom-components/map/map.component.ts

```diff
@@ -29,86 +30,81 @@ export class MapComponent implements OnInit, OnDestroy {
   ngOnInit(): void {
     this.mapView();
     const subscription$ = this.mapService.city$.subscribe((city) => {
-      this.onInitOffers();
       this.city = city;
+      this.onChangeOffers();
     });
     this.subscription.push(subscription$);
 
     const offersSubscription$ = this.mapService.offer$.subscribe((offer) => {
-      if (offer == null) this.onInitOffers();
-      else this.onChangeMarkerOffers(offer);
+      if (offer == null) this.onChangeOffers();
+      else this.onClickItem(offer);
     });
     this.subscription.push(offersSubscription$);
 
     const officeSubscription$ = this.mapService.office$.subscribe((office) => {
-      if (office !== null) this.onChangeMarkerOffice(office);
+      if (office !== null) this.onChangeOffice(office);
     });
     this.subscription.push(officeSubscription$);
+
+    const vendorSubscription$ = this.mapService.vendor$.subscribe((vendor) => {
+      if (vendor !== null) this.onClickItem(vendor);
+    });
+    this.subscription.push(vendorSubscription$);
   }
 
-  onChangeMarkerOffice(office: Office): void {
-    this.vendorRequest$.unsubscribe();
+  onChangeOffice(office: Office): void {
+    this.offerRequest$.unsubscribe();
     const markers = this.initOfficeMarkers(office);
     this.markerAll.clearLayers();
     this.markerAll.addLayers(markers);
     this.map.addLayer(this.markerAll);
   }
 
-  onInitOffers(): void {
-    this.vendorRequest$ = this.offerService
+  onChangeOffers(): void {
+    this.offerRequest$ = this.offerService
       .getOffers({ city: this.city })
       .subscribe((offers) => {
-        const markers = this.initOffersMarkers(offers, this.city);
+        const markers = this.initOffersMarkers(offers);
         this.markerAll.clearLayers();
         this.markerAll.addLayers(markers);
         this.map.addLayer(this.markerAll);
       });
   }
 
-  onChangeMarkerOffers(offer: Offer): void {
-    this.vendorRequest$.unsubscribe();
-    const markers = this.initOfferMarkers(offer);
+  onClickItem(data: Offer | Vendor): void {
+    this.offerRequest$.unsubscribe();
+    const markers = this.initMarkers(data);
     this.markerAll.clearLayers();
     this.markerAll.addLayers(markers);
     this.map.addLayer(this.markerAll);
   }
 
-  private mapView() {
-    const tiles = L.tileLayer(
-        'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
-        {
-          maxZoom: 18,
-          attribution:
-            '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, Points &copy 2012 LINZ'
-        }
-      ),
-      latlng = L.latLng(0, 0);
-    this.map = L.map('map', { center: latlng, zoom: 11, layers: [tiles] });
-  }
-
-  private initOffersMarkers(offers: Offer[], city: string) {
-    this.mapService.getCityView(city).then((data) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private initMarkers(data: any) {
```

Comment:
"data: any" is not the best decision

Reply:
Run the code and try to understand why we used "any"